### PR TITLE
Fixed metadata propagation in Dataframe.idxmax and Dataframe.idxmin #…

### PR DESCRIPTION
**Pull request by SomtochiUmeh**
…28283

- [x] contributes to #28283  calling .__finalize__ on Dataframe.idxmax, idxmin

### CODE REVIEW

File: pandas/tests/generic/test_finalize.py [hunk 1]
```diff
@@ -227,12 +227,10 @@
...
     ),
     pytest.param(
-        (pd.DataFrame, frame_data, operator.methodcaller("idxmin")),
-        marks=not_implemented_mark,
+        (pd.DataFrame, frame_data, operator.methodcaller("idxmin"))
     ),
     pytest.param(
-        (pd.DataFrame, frame_data, operator.methodcaller("idxmax")),
-        marks=not_implemented_mark,
+        (pd.DataFrame, frame_data, operator.methodcaller("idxmax"))
```
Comment (mzeitlin11) [inline pandas/tests/generic/test_finalize.py:233]: I think you can also remove the `pytest.param` call here now that the mark is not added

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @SomtochiUmeh! One small comment

**Comment by mzeitlin11** [comment]
Also, please see https://pandas.pydata.org/docs/development/contributing_codebase.html#pre-commit for how to fix that CI failure

File: pandas/tests/generic/test_finalize.py (hunk 1 above)
Comment (SomtochiUmeh) [reply pandas/tests/generic/test_finalize.py:233]: Noted. Thanks!

**Review by mzeitlin11 — APPROVED** [review]
LGTM, thanks @SomtochiUmeh!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @SomtochiUmeh

**Pull request merged**